sentry_sdk/scope.py

```diff
@@ -146,21 +160,178 @@ class Scope(object):
         "_force_auto_session_tracking",
         "_profile",
         "_propagation_context",
+        "client",
+        "original_scope",
+        "_type",
     )
 
-    def __init__(self):
-        # type: () -> None
+    def __init__(self, ty=None, client=None):
+        # type: (Optional[ScopeType], Optional[sentry_sdk.Client]) -> None
+        self._type = ty
+        self.original_scope = None  # type: Optional[Scope]
+
         self._event_processors = []  # type: List[EventProcessor]
         self._error_processors = []  # type: List[ErrorProcessor]
 
         self._name = None  # type: Optional[str]
         self._propagation_context = None  # type: Optional[Dict[str, Any]]
 
+        self.client = NoopClient()  # type: sentry_sdk.client.BaseClient
+
+        if client is not None:
+            self.set_client(client)
+
         self.clear()
 
         incoming_trace_information = self._load_trace_data_from_env()
         self.generate_propagation_context(incoming_data=incoming_trace_information)
 
+    @classmethod
+    def get_current_scope(cls):
+        # type: () -> Scope
+        """
+        Returns the current scope.
+
+        .. versionadded:: X.X.X
+        """
+        current_scope = _current_scope.get()
+        if current_scope is None:
+            current_scope = Scope(ty=ScopeType.CURRENT)
+            _current_scope.set(current_scope)
+
+        return current_scope
+
+    @classmethod
+    def get_isolation_scope(cls):
+        # type: () -> Scope
+        """
+        Returns the isolation scope.
+
+        .. versionadded:: X.X.X
+        """
+        isolation_scope = _isolation_scope.get()
+        if isolation_scope is None:
+            isolation_scope = Scope(ty=ScopeType.ISOLATION)
+            _isolation_scope.set(isolation_scope)
+
+        return isolation_scope
+
+    @classmethod
+    def get_global_scope(cls):
+        # type: () -> Scope
+        """
+        Returns the global scope.
+
+        .. versionadded:: X.X.X
+        """
+        global _global_scope
+        if _global_scope is None:
+            _global_scope = Scope(ty=ScopeType.GLOBAL)
+
+        return _global_scope
+
+    @classmethod
+    def _merge_scopes(cls, additional_scope=None, additional_scope_kwargs=None):
```

Comment:
What is the purpose of merging scopes?

Reply:
To apply all relevant data to the outgoing event we merge all data from the global, isolation and current scope and then apply this to the event.